Il2CppInspector.Common/Reflection/TypeInfo.cs

```diff
@@ -517,7 +521,17 @@ private class TypeArgumentsComparer : EqualityComparer<TypeInfo[]>
         public bool HasElementType => ElementType != null;
 
         private readonly TypeRef[] implementedInterfaceReferences;
-        public IEnumerable<TypeInfo> ImplementedInterfaces => implementedInterfaceReferences.Select(x => x.Value);
+        public IEnumerable<TypeInfo> ImplementedInterfaces {
+            get {
+                if (Definition != null)
+                    return implementedInterfaceReferences.Select(x => x.Value);
+                if (genericTypeDefinition != null)
+                    return genericTypeDefinition.ImplementedInterfaces.Select(t => t.SubstituteGenericArguments(genericArguments));
+                if (IsGenericParameter)
+                    return GetGenericParameterConstraints().Where(t => t.IsInterface);
```

Comment:
What if the interface constraint is generic? eg.

```cs
public class Foo<T> where T : IBar<T>
```

I think from the code here that it does resolve correctly but just wanted to check.

Reply:
Yeah, it'll work correctly. `IBar<T>` will be encoded as a regular TypeReference since `T` is known at compile-time.